central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -54,23 +87,152 @@ func (m *managerImpl) ProcessComplianceOperatorInfo(ctx context.Context, complia
+		return nil, errors.Wrapf(err, "Unable to create scan configuration named %q.", scanRequest.GetScanName())
+	}
+	if found {
+		return nil, errors.Errorf("Scan configuration named %q already exists.", scanRequest.GetScanName())
+	}
+
+	scanRequest.Id = uuid.NewV4().String()
+	scanRequest.CreatedTime = types.TimestampNow()
+	err = m.scanSettingDS.UpsertScanConfiguration(ctx, scanRequest)
+	if err != nil {
+		log.Error(err)
+		return nil, errors.Errorf("Unable to save scan configuration named %q.", scanRequest.GetScanName())
+	}
+
+	var profiles []string
+	for _, profile := range scanRequest.GetProfiles() {
+		profiles = append(profiles, profile.GetProfileName())
+	}
+
+	m.requestsLock.Lock()
+	defer m.requestsLock.Unlock()
+
+	for _, cluster := range clusters {
+		// id for the request message to sensor
+		id := uuid.NewV4().String()
+
+		sensorMessage := &central.MsgToSensor{
+			Msg: &central.MsgToSensor_ComplianceRequest{
+				ComplianceRequest: &central.ComplianceRequest{
+					Request: &central.ComplianceRequest_ApplyScanConfig{
+						ApplyScanConfig: &central.ApplyComplianceScanConfigRequest{
+							Id: id,
+							ScanRequest: &central.ApplyComplianceScanConfigRequest_ScheduledScan_{
+								ScheduledScan: &central.ApplyComplianceScanConfigRequest_ScheduledScan{
+									ScanSettings: &central.ApplyComplianceScanConfigRequest_BaseScanSettings{
+										ScanName:       scanRequest.GetScanName(),
+										StrictNodeScan: true,
+										Profiles:       profiles,
+									},
+									Cron: cron,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		err := m.sensorConnMgr.SendMessage(cluster, sensorMessage)
+		var status string
+		if err != nil {
+			status = err.Error()
+			log.Errorf("error sending compliance scan config to cluster %q: %v", cluster, err)
+		} else {
+			// Request was not rejected so add it to map awaiting response
+			m.requests[id] = clusterScan{
+				clusterID: cluster,
+				scanID:    scanRequest.GetId(),
+			}
+		}
+
+		// Update status in DB
+		err = m.scanSettingDS.UpdateClusterStatus(ctx, scanRequest.GetId(), cluster, status)
+		if err != nil {
+			log.Error(err)
+			return nil, errors.Errorf("Unable to save scan configuration status for scan named %q.", scanRequest.GetScanName())
+		}
+	}
+
+	return scanRequest, nil
+}
+
+// HandleScanRequestResponse processes response of compliance scan configuration from a sensor.
+func (m *managerImpl) HandleScanRequestResponse(ctx context.Context, requestID string, clusterID string, responsePayload string) error {
+	if !features.ComplianceEnhancements.Enabled() {
+		return nil
+	}
+
+	m.requestsLock.Lock()
+	defer m.requestsLock.Unlock()
+
+	var scanID string
+	if clusterScanData, found := m.requests[requestID]; found {
+		if clusterScanData.clusterID != clusterID {
+			return errors.Errorf("Cluster mismatch for request %q", requestID)
+		}
+		scanID = clusterScanData.scanID
+	} else {
+		return errors.Errorf("Unable to find request %q", requestID)
+	}
```

Comment:
This does not take into consideration that Central can restart after sending the request and before receiving the response.

Reply:
@md2119 Right.  I figured we would take care of that with the sync of configurations under https://issues.redhat.com/browse/ROX-18711.  At that time we would get the latest status of any configurations on the clusters, and apply any configurations that need it.  Does that sound reasonable?  I should probably put in a comment about that.

Reply:
Maybe add a TODO to this code block